### DISCUSSION TITLE
Eternal token

### DISCUSSION THREAD

**Discussion by AVANG003** [keycloak/keycloak#32917] [Q&A]
How to make eternal tokens in Keycloak for some client?

**Comment by codespearhead** [comment]
What are you trying to accomplish?

Because the whole point of OAuth2 is to make the tokens short-lived for security reasons.

If you're dealing with a Machine-to-Machine application, you should use the [Client Credentials Flow](https://medium.hexadefence.com/keycloak-integration-with-a-spring-boot-api-b12300d8fe75), because its credentials (client ID and client secret) don't expire:

> The Client Credentials Flow (defined in [OAuth 2.0 RFC 6749, section 4.4](https://tools.ietf.org/html/rfc6749#section-4.4)) involves an application exchanging its application credentials, such as client ID and client secret, for an access token.
> 
> This flow is best suited for Machine-to-Machine (M2M) applications, such as CLIs, daemons, or backend services, because the system must authenticate and authorize the application instead of a user.
> [[Source](https://auth0.com/docs/get-started/authentication-and-authorization-flow/client-credentials-flow)]
